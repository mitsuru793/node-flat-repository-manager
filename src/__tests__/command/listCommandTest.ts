import { execSync } from 'child_process'
import { mkdirSync } from 'fs'
import tmpDir = require('temp-dir')
import rimraf = require('rimraf')

function cli(args: string = ''): string {
  const cmd = `NODE_ENV=debug ts-node ${__dirname}/../../index.ts --projectsRoot ${tmpDir} ${args}`
  return execSync(cmd).toString()
}

describe('list command', () => {
  beforeEach(() => {
    rimraf.sync(`${tmpDir}/*`)

    const projectDirs = ['php-hello', 'php-world', 'node-hi']
    projectDirs.forEach(p => {
      mkdirSync(`${tmpDir}/${p}`)
    })
  })

  it('lists success', async () => {
    const out = cli('list')
    expect(out).toMatchSnapshot()
  })
})
