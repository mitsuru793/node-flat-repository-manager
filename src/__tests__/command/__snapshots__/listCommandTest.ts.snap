// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list command lists success 1`] = `
"/private/var/folders/p6/zm1pft7j33jbbtg64ytbxk7r0000gn/T
node
  * hi
php
  * hello
  * world
"
`;
